mm2src/coins/utxo/utxo_common.rs

```diff
@@ -5277,12 +5302,15 @@ pub fn derive_htlc_key_pair(coin: &UtxoCoinFields, _swap_unique_data: &[u8]) ->
 }
 
 #[inline]
-pub fn derive_htlc_pubkey(coin: &dyn SwapOps, swap_unique_data: &[u8]) -> [u8; 33] {
-    coin.derive_htlc_key_pair(swap_unique_data)
-        .public_slice()
-        .to_vec()
-        .try_into()
-        .expect("valid pubkey length")
+pub fn derive_htlc_pubkey(coin: &UtxoCoinFields, swap_unique_data: &[u8]) -> [u8; 33] {
+    match coin.priv_key_policy {
+        PrivKeyPolicy::WalletConnect { public_key, .. } => public_key.0,
+        _ => derive_htlc_key_pair(coin, swap_unique_data)
```

Comment:
Don't like the wildcard pattern, please use an exhaustive list.

Reply:
[6d4e79c](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2566/commits/6d4e79c2889fdbbecac5cb44e64be19d41a4a5d5)